examples/sparse/graphsage.py

```diff
@@ -0,0 +1,230 @@
+                sampled_mat = A.sample(1, fanout, ids=src, replace=True)
+                # Compact the matrix
+                compacted_mat, row_ids = sampled_mat.compact(0)
+                sampled_mats.insert(0, compacted_mat)
+                src = row_ids
+
+            x = ndata["feat"][src]
+            y = ndata["label"][seeds]
+            ys.append(y)
+            y_hats.append(model(sampled_mats, x))
+
+    return MF.accuracy(
+        torch.cat(y_hats),
+        torch.cat(ys),
+        task="multiclass",
+        num_classes=num_classes,
+    )
+
+
+def validate(device, A, ndata, dataset, model, batch_size):
+    model.eval()
+    inf_id = dataset.test_idx.to(device)
+    inf_dataloader = torch.utils.data.DataLoader(inf_id, batch_size=batch_size)
+    acc = evaluate(model, A, inf_dataloader, ndata, dataset.num_classes)
+    return acc
+
+
+def train(device, A, ndata, dataset, model):
+    # Create sampler & dataloader.
+    train_idx = dataset.train_idx.to(device)
+    val_idx = dataset.val_idx.to(device)
+
+    train_dataloader = torch.utils.data.DataLoader(
+        train_idx, batch_size=1024, shuffle=True
+    )
+    val_dataloader = torch.utils.data.DataLoader(val_idx, batch_size=1024)
+
+    opt = torch.optim.Adam(model.parameters(), lr=1e-3, weight_decay=5e-4)
+
+    fanouts = [10, 10, 10]
+    for epoch in range(10):
+        model.train()
+        total_loss = 0
+        for it, (seeds) in enumerate(train_dataloader):
+            src = seeds
+            sampled_mats = []
+            for fanout in fanouts:
+                # Sampling neighbor
+                sampled_mat = A.sample(1, fanout, ids=src, replace=True)
+                # Compact the matrix
+                compacted_mat, row_ids = sampled_mat.compact(0)
+                sampled_mats.insert(0, compacted_mat)
+                src = row_ids
+
+            x = ndata["feat"][src]
+            y = ndata["label"][seeds]
+            y_hat = model(sampled_mats, x)
+            loss = F.cross_entropy(y_hat, y)
+            opt.zero_grad()
+            loss.backward()
+            opt.step()
+            total_loss += loss.item()
+
+        acc = evaluate(model, A, val_dataloader, ndata, dataset.num_classes)
+        print(
+            "Epoch {:05d} | Loss {:.4f} | Accuracy {:.4f} ".format(
+                epoch, total_loss / (it + 1), acc.item()
+            )
+        )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="GraphSAGE")
+    parser.add_argument(
+        "--mode",
+        default="gpu",
+        choices=["cpu", "gpu"],
+        help="Training mode. 'cpu' for CPU training, "
+        "'gpu' for pure-GPU training.",
+    )
+    args = parser.parse_args()
+    if not torch.cuda.is_available():
+        args.mode = "cpu"
+    print(f"Training in {args.mode} mode.")
+
+    #####################################################################
+    # (HIGHLIGHT) Node classification task is a supervise learning task
```

Comment:
Is this the HIGHLIGHT of the example? I think the HIGHLIGHT should be sampling with sparse matrix.